abn/service.go

```diff
@@ -0,0 +1,142 @@
+// Package abn contains the implementation of the A/B/n service (via gRPC)
+package abn
+
+// service.go - entry point for A/B/n service
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"golang.org/x/sys/unix"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/dgraph-io/badger/v4"
+	pb "github.com/iter8-tools/iter8/abn/grpc"
+	"github.com/iter8-tools/iter8/base/log"
+	"github.com/iter8-tools/iter8/controllers"
+	"github.com/iter8-tools/iter8/storage"
+	"github.com/iter8-tools/iter8/storage/badgerdb"
+
+	// auth package is necessary to enable authentication with various cloud providers
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+)
+
+const (
+	// defaultMetricsPath is the default path of the persistent volume
+	defaultMetricsPath = "/metrics"
+)
+
+var (
+	// MetricsPath is the path of the persistent volume
+	metricsPath = defaultMetricsPath
+	// MetricsClient is the metrics client
+	MetricsClient storage.Interface
+)
+
+// newServer returns a new gRPC server
+func newServer() *abnServer {
+	return &abnServer{}
+}
+
+type abnServer struct {
+	pb.UnimplementedABNServer
+}
+
+// Lookup identifies a track that should be used for a given user
+// This method is exposed to gRPC clients
+func (server *abnServer) Lookup(ctx context.Context, appMsg *pb.Application) (*pb.Session, error) {
+	log.Logger.Tracef("Lookup called for application=%s, user=%s", appMsg.GetName(), appMsg.GetUser())
+	defer log.Logger.Trace("Lookup completed")
+
+	_, track, err := lookupInternal(
+		appMsg.GetName(),
+		appMsg.GetUser(),
+		&controllers.AllRoutemaps,
+	)
+
+	if err != nil {
+		log.Logger.Warn("Lookup failed: ", err)
+		return nil, err
+	}
+
+	if track == nil {
+		log.Logger.Warn("lookup returned nil")
```

Comment:
Doesn't match the other logger statements

```suggestion
		log.Logger.Warn("Lookup returned nil")
```

Reply:
done